base/applications/mspaint/toolsettings.cpp

```diff
@@ -25,7 +25,10 @@ LRESULT CToolSettingsWindow::OnCreate(UINT nMsg, WPARAM wParam, LPARAM lParam, W
 
 LRESULT CToolSettingsWindow::OnVScroll(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
 {
-    zoomTo(125 << trackbarZoom.SendMessage(TBM_GETPOS, 0, 0), 0, 0);
+    if (!zoomTo(125 << trackbarZoom.SendMessage(TBM_GETPOS, 0, 0), 0, 0))
+    {
+        OnToolsModelZoomChanged(nMsg, wParam, lParam, bHandled);
```

Comment:
Just to be sure, this new line is for "Do not allow to move zoom slider above possible position." ?

Reply:
Exactly. Since `zoomTo` is failed, calling this function resets the slider to current zoom rate.